mmpose/datasets/datasets/body3d/body3d_mview_direct_campus_dataset.py

```diff
@@ -0,0 +1,627 @@
+
+    def __init__(self,
+                 ann_file,
+                 img_prefix,
+                 data_cfg,
+                 pipeline,
+                 dataset_info=None,
+                 test_mode=False):
+
+        if dataset_info is None:
+            warnings.warn(
+                'dataset_info is missing. '
+                'Check https://github.com/open-mmlab/mmpose/pull/663 '
+                'for details.', DeprecationWarning)
+            cfg = Config.fromfile('configs/_base_/datasets/campus.py')
+            dataset_info = cfg._cfg_dict['dataset_info']
+
+        super().__init__(
+            ann_file,
+            img_prefix,
+            data_cfg,
+            pipeline,
+            dataset_info=dataset_info,
+            test_mode=test_mode)
+
+        self.load_config(data_cfg)
+        self.ann_info['use_different_joint_weights'] = data_cfg.get(
+            'use_different_joint_weights', False)
+
+        self.db_size = self.num_cameras * len(
+            self.frame_range
+        ) if self.test_mode else self.num_cameras * self.num_train_samples
+        print(f'=> load {self.db_size} samples')
+
+    def load_config(self, data_cfg):
+        """Initialize dataset attributes according to the config.
+
+        Override this method to set dataset specific attributes.
+        """
+        self.num_joints = data_cfg['num_joints']
+        self.cam_list = data_cfg['cam_list']
+        self.num_cameras = data_cfg['num_cameras']
+        assert self.num_cameras == len(self.cam_list)
+        self.need_camera_param = True
+
+        self.frame_range = data_cfg['frame_range']
+
+        self.width = data_cfg.get('width', 360)
+        self.height = data_cfg.get('height', 288)
+        self.center = np.array((self.width / 2, self.height / 2),
+                               dtype=np.float32)
+        self.scale = self._get_scale((self.width, self.height))
+
+        root_id = data_cfg.get('root_id', [11, 12])
+        self.root_id = [root_id] if isinstance(root_id, int) else root_id
+
+        self.max_nposes = data_cfg.get('max_nposes', 10)
+        self.min_nposes = data_cfg.get('min_nposes', 1)
+        self.num_train_samples = data_cfg.get('num_train_samples', 3000)
+        self.maximum_person = data_cfg.get('maximum_person', 10)
+
+        self.cam_file = data_cfg.get(
+            'cam_file', osp.join(self.img_prefix, 'calibration_campus.json'))
+
+        self.test_pose_db_file = data_cfg.get(
+            'test_pose_db_file',
+            osp.join(self.img_prefix, 'pred_campus_maskrcnn_hrnet_coco.pkl'))
+
+        self.train_pose_db_file = data_cfg.get(
+            'train_pose_db_file',
+            osp.join(self.img_prefix, 'panoptic_training_pose.pkl'))
+
+        self.gt_pose_db_file = data_cfg.get(
+            'gt_pose_db_file', osp.join(self.img_prefix, 'actorsGT.mat'))
+
+        self._get_db()
+
+    def _get_scale(self, raw_image_size):
```

Comment:
How about moving these general functions to the base class?

Reply:
Good suggestion, I will move them to the base class.